compiler/rustc_infer/src/infer/opaque_types.rs

```diff
@@ -645,3 +657,93 @@ fn may_define_opaque_type(tcx: TyCtxt<'_>, def_id: LocalDefId, opaque_hir_id: hi
     );
     res
 }
+
+/// Checks that the item may register hidden types for the
+/// opaque type, if the opaque type shows up in its signature.
+#[instrument(level = "debug", skip(tcx), ret)]
+pub fn may_define_impl_trait_in_assoc_ty_modulo_sig<'tcx>(
```

Comment:
Can you explain what you mean by "modulo sig" here?

Reply:
Like this seems like it should be called something more like "`get_sig_of_item_that_is_trying_to_define_impl_trait_in_assoc_ty`" but worded more gracefully.